### PR TITLE
Photozsystematics

### PR DESCRIPTION
**Pull request by zdu863**
Based on this [script](https://gist.github.com/MarkusMichaelRau/8ee36f06a34a74abda7b3853be2b60ba), the PR adds a systematic through convolving a redshift distribution with a corresponding error distribution according to [this paper](https://arxiv.org/pdf/1002.3383.pdf) Eq. 5.

### CODE REVIEW

File: firecrown/ccl/systematics/pz.py [hunk 1]
```diff
@@ -37,3 +39,54 @@ def apply(self, cosmo, params, source):
...
+        ----------
+        cosmo : pyccl.Cosmology
+            A pyccl.Cosmology object.
+        params : dict
+            A dictionary mapping parameter names to their current values.
+        source : a source object
+            The source to which apply the shift.
+        """
+        zorig, N = (source.z_, len(source.z_))
+        zpdf_orig = source.dndz_/np.sum(source.dndz_*source.z_)
```
Comment (beckermr) [inline firecrown/ccl/systematics/pz.py]: This confuses me. z_ is the array of z values, not the bin spacing. Also, dndz is already normalized here.

File: firecrown/ccl/systematics/pz.py [hunk 2]
```diff
@@ -37,3 +39,54 @@ def apply(self, cosmo, params, source):
...
+        joint_distr = np.zeros((N, N))
```
Comment (beckermr) [inline firecrown/ccl/systematics/pz.py]: There is no reason to keep the full array of values around.

File: firecrown/ccl/systematics/tests/test_pz.py [hunk 3]
```diff
@@ -58,3 +59,29 @@ def test_photoz_shift_fill_nan():
     msk = z - 0.5 <= 0
     assert np.all(src.dndz_[msk] == 0)
     assert np.all(src.dndz_[~msk] != 0)
+
+
+def test_photoz_systematic_smoke():
```
Comment (beckermr) [inline firecrown/ccl/systematics/tests/test_pz.py]: This test doesn't actually test anything. You are setting an attribute and then comparing it to what you set it to.

**Comment by vitenti** [comment]
This PR contain code for applying a Gaussian distribution to a dndz. We are planning to have a separate object to describe dndz and this code may be used there in the future. I'm closing this PR and any useful code will be copied to the new structure.

**Pull request closed without merging**